core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -364,7 +379,22 @@ private[spark] class CoarseMesosSchedulerBackend(
   }
 
   override def stop() {
-    super.stop()
+    // Make sure we're not launching tasks during shutdown
+    stateLock.synchronized {
+      if (!stopCalled.compareAndSet(false, true)) {
```

Comment:
I don't think we need to make stopCalled atomic boolean since we already have a lock around this.